keras2onnx/ke2onnx/simplernn.py

```diff
@@ -274,94 +274,31 @@ def build_output(scope, operator, container, output_names, bidirectional=False):
-                                              ('_b', oopb.int64, np.array([-1], dtype='int64')),
-                                              ('_c', oopb.int64, np.array([hidden_size], dtype='int64'))
-                                              ],
-                                             input_name + '_output_seq_shape', axis=0)
-                rnn_y_fixed = oopb.add_node('Reshape',
-                                                  [rnn_y,
-                                                   shape_tensor
-                                                   ],
-                                                  input_name + '_output_seq_shape_1')
-
+                # Transpose RNN Y with shape (T, D, N, C) into (T, N, D, C)
+                apply_transpose(scope, rnn_y, lstm_out, container, perm=[0, 2, 1, 3])
             if merge_concat:
-                # In this case, only one Keras output with shape (N, T, 2 * C') should be produced
-
-                # Transpose ONNX RNN Y with shape (T, D, N, C') into (T, N, D, C')
-                transposed_y = _name('Y_transposed')
-                apply_transpose(scope, rnn_y_fixed, transposed_y, container, perm=[2, 0, 1, 3])
-
-                # Change shape (T, N, D, C') to (N, T, D * C') to meet Keras spec
-                if is_static_shape:
-                    apply_reshape(scope, transposed_y, operator.outputs[0].full_name, container,
-                                  desired_shape=[-1, seq_length, 2 * hidden_size])
-                else:
-                    attrs = {'axis': 0}
-                    shape_tensor_2 = oopb.add_node('Concat',
-                                                   [('_a', oopb.int64, np.array([-1], dtype='int64')),
-                                                    seq_dim,
-                                                    ('_b', oopb.int64, np.array([2 * hidden_size], dtype='int64'))
-                                                    ],
-                                                   input_name + '_output_seq_shape_2', **attrs)
-                    shape_tensor_output = oopb.add_node('Reshape',
-                                                        [transposed_y,
-                                                         shape_tensor_2
-                                                         ],
-                                                        input_name + '_output_merge_concat')
-                    apply_identity(scope, shape_tensor_output, operator.outputs[0].full_name, container)
+                # In this case, only one Keras output with shape (N, T, 2 * C') should be produced.
+                # ( T, N, 2*C ) if it was time major.
+                apply_reshape(scope, lstm_out, operator.outputs[0].full_name, container,
+                              desired_shape=[0, 0, 2 * hidden_size])
             else:
                 # If merge_mode=None, two tensors should be generated. The first/second tensor is the output of
```

Comment:
For the case `merge_concat=False`, we have `apply_reshape` to handle static/dynamic shapes. Now seems that the new code does not need that any more. It only `apply_reshape` for `merge_concat=True`. Why we have difference here?

Reply:
When merge_concat is false or None, we need to produce two tensors as output - one for forward part and one for backward. The output of bidirectional lstm is of shape [seq_len, 2, batch_size, input_size]. 
**In case of time_major=true**
1. We first apply split two split it across the num_direction axis: [seq_len, 1, batch_size, input_size] and [ seq_len, 1, batch_size, input_size]
2. Then we squeeze in axis 1, to produce two tensors of shape  [seq_len, batch_size, input_size] and [ seq_len, batch_size, input_size]

**In case of time_major=false**
1. We do the transpose to make it batch major: [batch_size, 2, seq_len, input_size]
2. We apply split two split it across the num_direction axis: [batch_size, 1, seq_len, input_size] and [batch_size, 1, seq_len, input_size]
3. Then we squeeze in axis 1, to produce two tensors of shape  [batch_size, seq_len, input_size] and [batch_size, seq_len, input_size]